packages/timeline-state-resolver/src/integrations/casparCG/index.ts

```diff
@@ -168,6 +168,7 @@ export class CasparCGDevice extends DeviceWithState<State, DeviceOptionsCasparCG
 			this._ccg.disconnect()
 			this._ccg.onDisconnected = () => {
 				resolve(true)
+				this._ccg.removeAllListeners()
```

Comment:
I think that there is actually a potential problem here. If the caspar connection happens to be disconnected here, the onDisconnected won't be called (see https://github.com/SuperFlyTV/casparcg-connection/blob/5.0.0/src/CasparCG.ts#L2206). So we should perhaps do this here:
```

this._ccg.disconnect()
this._ccg.onDisconnected = () => {
  resolve(true)
  this._ccg.removeAllListeners()
}
if (!this._ccg.connected) this._ccg.onDisconnected()
```